src/bespokelabs/curator/request_processor/litellm_online_request_processor.py

```diff
@@ -0,0 +1,301 @@
+                    await self.client.chat.completions.create_with_completion(
+                        **request.api_specific_request,
+                        response_model=request.prompt_formatter.response_format,
+                        timeout=60.0,
+                    )
+                )
+                response_message = (
+                    response.model_dump() if hasattr(response, "model_dump") else response
+                )
+            else:
+                completion_obj = await litellm.acompletion(
+                    **request.api_specific_request, timeout=60.0
+                )
+                response_message = completion_obj["choices"][0]["message"]["content"]
+
+            # Extract token usage
+            usage = completion_obj.usage if hasattr(completion_obj, "usage") else {}
+            token_usage = TokenUsage(
+                prompt_tokens=usage.prompt_tokens,
+                completion_tokens=usage.completion_tokens,
+                total_tokens=usage.total_tokens,
+            )
+
+            # Calculate cost using litellm
+            cost = litellm.completion_cost(completion_response=completion_obj.model_dump())
+
+            # Create and save response immediately
+            generic_response = GenericResponse(
+                response_message=response_message,
+                response_errors=None,
+                raw_request=request.api_specific_request,
+                raw_response=completion_obj.model_dump(),
+                generic_request=request.generic_request,
+                created_at=request.created_at,
+                finished_at=datetime.datetime.now(),
+                token_usage=token_usage,
+                response_cost=cost,
+            )
+
+            await self.append_generic_response(generic_response, save_filepath)
+            status_tracker.num_tasks_in_progress -= 1
+            status_tracker.num_tasks_succeeded += 1
+            status_tracker.pbar.update(1)
+
+        except Exception as e:
+            logger.error(f"Error in API request: {e}")
+            status_tracker.num_other_errors += 1
+            request.result.append(e)
+
+            if request.attempts_left > 0:
+                request.attempts_left -= 1
+                retry_queue.put_nowait(request)
+            else:
+                generic_response = GenericResponse(
+                    response_message=None,
+                    response_errors=[str(e) for e in request.result],
+                    raw_request=request.api_specific_request,
+                    raw_response=None,
+                    generic_request=request.generic_request,
+                    created_at=request.created_at,
+                    finished_at=datetime.datetime.now(),
+                )
+                await self.append_generic_response(generic_response, save_filepath)
+                status_tracker.num_tasks_in_progress -= 1
+                status_tracker.num_tasks_failed += 1
```

Comment:
should this be in the base class? We should also logger.ERROR when we hit max attempts

Reply:
the part in the try is in the implemented class
the try and except in the base class